pkg/git/git_test.go

```diff
@@ -29,7 +29,31 @@ import (
 
 const fileMode = 0755 // rwxr-xr-x
 
+func withTemporaryGitConfig(t *testing.T) func() {
+	gitConfigDir := t.TempDir()
+	key := "GIT_CONFIG_GLOBAL"
+	t.Helper()
+	oldValue, envVarExists := os.LookupEnv(key)
+	if err := os.Setenv(key, filepath.Join(gitConfigDir, "config")); err != nil {
```

Comment:
Does the `os.Setenv` break the per-test isolation in any way?

Reply:
It shouldn't no, but if the run in parallel (which I don't think they are configured to do), it could yes.

Reply:
Note, that there is no way around it (except make sure they are not running in parallel). And this is also something we do elsewhere I am pretty sure 🙃

Reply:
Yes, it won't be an issue if they are not executed in parallel, but perhaps it's worth adding a comment about it in case someone tries to do so in future. Not something that was changed by this PR anyways, so no need to re-spin this.